pkg/apis/cluster/v1alpha1/types.go

```diff
@@ -116,6 +137,79 @@ type ClusterSpec struct {
 	// any resource that does not tolerate the Taint.
 	// +optional
 	Taints []corev1.Taint `json:"taints,omitempty"`
+
+	// ResourceModels is the list of resource modeling in this cluster. Each modeling quota can be customized by the user.
+	// Modeling name must be one of the following: cpu, memory, storage, ephemeral-storage.
+	// If the user does not define the modeling name and modeling quota, it will be the default model.
+	// The default model index from 0 to 11. Each default model's cpu quota is (2^index-1, 2^index], index >= 0
+	// Each default model's memory quota is also (2^index-1, 2^index], index >= 0, for example, model0 is like this:
+	// - index: 0
+	// ranges:
+	//   - name: "cpu"
+	//     min: 0 C
+	//     max: 1 C
+	//   - name: "memory"
+	//     min: 0 GB
+	//     max: 1 GB
+	//
+	// model10 is like this:
+	// - index: 10
+	// range:
+	//   - name: "cpu"
+	//     min: 512 C
+	//     max: 1024 C
+	//   - name: "memory"
+	//     min: 512 GB
+	//     max: 1024 GB
+	//
+	// model11 is like this:
+	// - index: 11
+	// range:
+	//   - name: "cpu"
+	//     min: 1024 C
+	//     max: MAXINT
+	//   - name: "memory"
+	//     min: 1024 GB
+	//     max: MAXINT
+	// +optional
+	ResourceModels []ResourceModel `json:"resourceModels,omitempty"`
+}
+
+// ResourceModel describes the modeling that you want to statistics.
+type ResourceModel struct {
+	// Grade is the index for the resource modeling.
+	// +optional
+	Grade int `json:"grade,omitempty"`
+
+	// Ranges describes the resource quota ranges.
+	// +optional
+	Ranges []ResourceModelItem `json:"ranges,omitempty"`
+}
+
+// ResourceModelItem describes the detail of each modeling quota that ranges from min to max.
+// Please pay attention, by default, the value of min can be obtained, and the value of max cannot be obtained.
+// For example, in an interval, min = 2, max =10 is set, which means the interval [2,10).
+// This rule ensure that all intervals have the same meaning. If the last interval is +∞,
+// it is definitely unreachable. Therefore, we define the right interval as the open interval.
+// For a valid interval, the value on the right is greater than the value on the left,
+// in other words, max must be greater than min.
+type ResourceModelItem struct {
+	// Name is the name for the resource that you want to categorize.
+	// +optional
+	Name ResourceName `json:"name,omitempty"`
+
+	// Min is the minimum amount of this resource represented by resource name。
+	// +optional
+	Min resource.Quantity `json:"min,omitempty"`
+
+	// Max is the maximum amount of this resource represented by resource name。
+	// Special Instructions, for the last ResourceModelItem, which no matter what Max value you pass,
+	// the meaning is infinite. Because for the last item,
+	// any ResourceModelItem's quota larger than Min will be classified to the last one.
```

Comment:
I basically agree with this that put all notes into the `last`(largest) model.
My concern is, from API point of view it might be counter-intuitive, for example:
I specified the last model is `(10c, 20c}`, why the nodes with `30c` are counted in this model ?

cc @kevin-wangzefeng for comments.

Reply:
I understand that users will be confused in this case. But in this case, if `30c` should not be assigned to any group, it is weird too. This is also not reasonable. Then `30c` is discarded and not counted? I think it might be better to join the last model in statistical calculations. What's your opinion? @kevin-wangzefeng